compiler/rustc_hir_analysis/src/astconv/errors.rs

```diff
@@ -1024,6 +1037,150 @@ impl<'tcx> dyn AstConv<'tcx> + '_ {
             Ok(())
         }
     }
+
+    pub fn report_prohibit_generics_error<'a>(
+        &self,
+        segments: impl Iterator<Item = &'a hir::PathSegment<'a>> + Clone,
+        prohibit_args: FxIndexSet<ProhibitGenericsArg>,
+        extend: impl Fn(&mut Diag<'_>),
+    ) {
```

Comment:
```suggestion
    ) -> ErrorGuaranteed {
```
and necessary changes for that to compile